src/SearchBundle/Bridge/ShopSearchSessionChameleonBridge.php

```diff
@@ -44,19 +48,23 @@ public function addSearch(array $searchRequest)
      */
     public function hasSearchedFor(array $searchRequest)
     {
+        $session = $this->getSession();
+        
+        if(null === $session) {
+            return false;
+        }
+        
         $searchKey = md5($this->getArrayAsString($searchRequest));
 
-        $searches = $this->session->get(ShopSearchSessionInterface::SESSION_KEY, array());
+        $searches = $session->get(ShopSearchSessionInterface::SESSION_KEY, array());
 
         return in_array($searchKey, $searches);
     }
 
     /**
      * @param array $data
-     *
-     * @return string
      */
-    private function getArrayAsString($data)
+    private function getArrayAsString($data): string
```

Comment:
Consequently, you can remove the whole annotation by specifying also the param type
```suggestion
    private function getArrayAsString(array $data): string
```